dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/query/Restrictions.java

```diff
@@ -134,7 +134,12 @@ public static Restriction isEmpty(String path) {
   }
 
   public static Disjunction query(Schema schema, String query) {
-    return or(schema, eq("id", query), eq("code", query), ilike("name", query, MatchMode.ANYWHERE));
+    Restriction name = ilike("name", query, MatchMode.ANYWHERE);
+    Restriction code = eq("code", query);
+    if (query.length() != 11) return or(schema, code, name);
```

Comment:
The name `query` is a bit confusing here for me. It sounds like it's the query param being checked for a length of 11. `query` to me implies it's the whole query string.

Reply:
It is called `query` in the URL and I am trying to avoid requiring mental mapping by calling the value for the parameter the same name hoping that this makes it easy to follow once there is a bit of background knowledge about the parameter itself.

Reply:
I added javadoc